modules/imgproc/test/test_resize_bitexact.cpp

```diff
@@ -152,4 +152,33 @@ TEST(Resize_Bitexact, Linear8U)
         }
 }
 
+static double checkNearest8U_vsNonExact(const Mat& src)
+{
+    Mat dstExact; cv::resize(src, dstExact, Size(), 2, 1, INTER_NEAREST_EXACT);
+    Mat dstNonExact; cv::resize(src, dstNonExact, Size(), 2, 1, INTER_NEAREST);
+
+    return cv::norm(dstExact, dstNonExact, NORM_INF);
+}
+TEST(Resize_Bitexact, Nearest8U_vsNonExact)
+{
+    int depths[] = {CV_8U, CV_16U, CV_32F, CV_64F};
```

Comment:
Please make this as test parameter:
- use `MatDepth` wrapped test type
- look for `INSTANTIATE_TEST_CASE_P`